src/App.js

```diff
@@ -1,25 +1,88 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import "./App.css";
+import LoginScreen from "./components/LoginScreen/LoginScreen";
+import Dashboard from "./components/Dashboard/Dashboard";
+import { Component } from "react";
+import { BrowserRouter as Router, Link, Switch, Route } from "react-router-dom";
+import "bootstrap/dist/css/bootstrap.min.css";
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isLoggedIn: false,
+      flightCodes: [],
+      squadronData: [],
+    };
+  }
+
+  async componentDidMount() {
+    const response = await fetch(
+      "https://us-central1-cc-dashboard-afb15.cloudfunctions.net/getFlightCodes"
+    );
+    const newFlightCodes = await response.json();
+    this.setState({ flightCodes: newFlightCodes });
+
+    let output = [];
+    await Promise.all(
+      await this.state.flightCodes.flights.map(async (code) => {
+        const airmenResponse = await fetch(
+          `https://us-central1-cc-dashboard-afb15.cloudfunctions.net/getAirmen/${code}`
+        );
+        const airmenData = await airmenResponse.json();
+        output.push({ flightName: code, flightData: airmenData });
+      })
+    );
+
+    this.setState({ squadronData: output });
+  }
+
+  handleLogin = (event) => {
+    this.setState({ isLoggedIn: true });
+  };
+  handleLogout = (event) => {
+    this.setState({ isLoggedIn: false });
+  };
```

Comment:
It's probably a good idea to bind these in the constructor after the assignment of this.state like so:
`this.state = {
...
};  
this.handleLogin = this.handleLogin.bind(this);
this.handleLogout = this.handleLogout.bind(this);
`

Since we pass these down as props to the Dashboard component, we want to make sure that the stateful (i.e. Class) component is what is referenced by them (or you can get a "this.setState is not a function" error when they get called in a context where the this value doesn't align to a Class component - even if it does, you can end up setting the state of the wrong Class component because this will defer to the context of the component in which the method is called)